AmplifyPlugins/Analytics/Tests/AnalyticsHostApp/AWSPinpointAnalyticsPluginIntegrationTests/AWSPinpointAnalyticsPluginIntegrationTests.swift

```diff
@@ -145,6 +146,87 @@ class AWSPinpointAnalyticsPluginIntergrationTests: XCTestCase {
+                let event = BasicAnalyticsEvent(name: "eventName" + String(eventNumber), properties: properties)
+                Amplify.Analytics.record(event: event)
+                await recordExpectation.fulfill()
+            }
+        }
+
+        await waitForExpectations([recordExpectation], timeout: TestCommonConstants.networkTimeout)
+    }
+    
+    /// - Given: Analytics plugin configured with valid configuration
+    /// - When: 50 different events with 20 attributes are recorded simultaneously
+    /// - Then: Operations are successful
+    func testMultipleLargeRecordEventAndFlush() async {
+        let onlineExpectation = expectation(description: "Device is online")
+        let networkMonitor = NWPathMonitor()
+        networkMonitor.pathUpdateHandler = { newPath in
+            if newPath.status == .satisfied {
+                onlineExpectation.fulfill()
+            }
+        }
+        networkMonitor.start(queue: DispatchQueue(label: "AWSPinpointAnalyticsPluginIntergrationTests.NetworkMonitor"))
+        
+        wait(for: [onlineExpectation], timeout: TestCommonConstants.networkTimeout)
+        
+        let recordExpectation = asyncExpectation(description: "Records are successfully recorded",
+                                                 expectedFulfillmentCount: concurrencyLimit)
+        for eventNumber in 0...concurrencyLimit {
+            let properties = ["eventPropertyStringKey1": "eventProperyStringValue1",
+                              "eventPropertyStringKey2": "eventProperyStringValue2",
+                              "eventPropertyStringKey3": "eventProperyStringValue3",
+                              "eventPropertyStringKey4": "eventProperyStringValue4",
+                              "eventPropertyStringKey5": "eventProperyStringValue5",
+                              "eventPropertyIntKey1": 123,
+                              "eventPropertyIntKey2": 123,
+                              "eventPropertyIntKey3": 123,
+                              "eventPropertyIntKey4": 123,
+                              "eventPropertyIntKey5": 123,
+                              "eventPropertyDoubleKey1": 12.34,
+                              "eventPropertyDoubleKey2": 12.34,
+                              "eventPropertyDoubleKey3": 12.34,
+                              "eventPropertyDoubleKey4": 12.34,
+                              "eventPropertyDoubleKey5": 12.34,
+                              "eventPropertyBoolKey1": true,
+                              "eventPropertyBoolKey2": true,
+                              "eventPropertyBoolKey3": true,
+                              "eventPropertyBoolKey4": true,
+                              "eventPropertyBoolKey5": true] as [String: AnalyticsPropertyValue]
+            let event = BasicAnalyticsEvent(name: "eventName" + String(eventNumber), properties: properties)
+            Amplify.Analytics.record(event: event)
+            await recordExpectation.fulfill()
```

Comment:
Shouldn't these lines be wrapped inside a `Task` as well, like in `testMultipleRecordEvent`?
Otherwise they won't be recorded **simultaneously** as stated in the description.

Reply:
thanks for catching this, fixed it!